src/hex.rs

```diff
@@ -63,78 +64,59 @@ impl<T: Hash> FromHex for T {
 
 /// Iterator over a hex-encoded string slice which decodes hex and yields bytes.
 pub struct HexIterator<'a> {
-    /// The slice whose first two characters will be decoded to yield the next byte
-    sl: &'a str
+    /// The `Chars` iterator whose first two characters will be decoded to yield
+    /// the next byte.
+    chars: Chars<'a>,
 }
 
 impl<'a> HexIterator<'a> {
-    /// Constructs a new `HexIterator` from a string slice. If the string is of odd length it
-    /// returns an error.
+    /// Constructs a new `HexIterator` from a string slice. If the string is of
+    /// odd length it returns an error.
```

Comment:
Nice! Someone else who likes comments to be within 80 columns! Do you do this manually or do you have a Vim plugin that allows different textwidths for code vs comments?

Reply:
Haha I use emacs and Meta-q seems to just auto-align the current comment block you're in :grin: